openedx/adg/lms/applications/admin.py

```diff
@@ -72,3 +81,284 @@ class BusinessLineAdmin(admin.ModelAdmin):
+
+    def location(self, obj):
+        user_profile = UserProfile.objects.get(user=obj.user)
+        return '{city}, {country}'.format(city=user_profile.city, country=user_profile.country)
+
+    def linkedIn_profile(self, obj):
+        return format_html('<a href={url}>{url}</a>', url=obj.linkedin_url)
+
+    def is_saudi_national(self, obj):
+        extended_user_profile = ExtendedUserProfile.objects.get(user=obj.user)
+        if extended_user_profile.is_saudi_national:
+            return 'Yes'
+        return 'No'
+    is_saudi_national.short_description = 'Are you a Saudi National?'
+
+    def gender(self, obj):
+        """
+        Return gender of applicant
+        """
+        user_profile = UserProfile.objects.get(user=obj.user)
+        if user_profile.gender == 'm':
+            return 'Man'
+        elif user_profile.gender == 'f':
+            return 'Woman'
+        else:
+            return 'Prefer not to answer'
+
+    def phone_number(self, obj):
+        return UserProfile.objects.get(user=obj.user).phone_number
+
+    def date_of_birth(self, obj):
+        return ExtendedUserProfile.objects.get(user=obj.user).birth_date.strftime('%d %B %Y')
+
+    def applying_to(self, obj):
+        return obj.business_line
+
+    def resume_display(self, obj):
+        return display_file(obj.resume)
+    resume_display.short_description = 'Resume'
+
+    def cover_letter_display(self, obj):
+        return display_file(obj.cover_letter_file)
+    cover_letter_display.short_description = 'Cover Letter'
+
+    def prerequisites(self, obj):
+        """
+        Return the html for course names of all prereq courses and applicant's respective scores in those courses
+        """
+        html_for_score = '<p>{course_name}: <b>{course_percentage}%</b></p>'
+        final_html = ''
+        prereq_scores = obj.prereq_course_scores
+        for prereq_score in prereq_scores:
+            final_html += html_for_score.format(
+                course_name=prereq_score.course_name,
+                course_percentage=prereq_score.course_percentage
+            )
+
+        return format_html(final_html)
+
+    def get_fieldsets(self, request, obj=None):
+        basic_info_fields = ['email', 'location']
+        if obj.linkedin_url:
+            basic_info_fields.append('linkedIn_profile')
+
+        fieldsets = [
+            (None, {
+                'fields': tuple(basic_info_fields)
+            }),
+            ('APPLICANT INFORMATION', {
+                'fields': (
+                        'is_saudi_national', 'gender', 'phone_number', 'date_of_birth', 'organization', 'applying_to'
+                ),
+            }),
+        ]
+
+        file_fields = []
+        fieldset_for_files = None
+
+        if obj.is_cover_letter_provided and obj.resume:
+            fieldset_for_files = RESUME_AND_COVER_LETTER
+            file_fields.append('resume')
+            if obj.cover_letter_file:
```

Comment:
The property `is_cover_letter_file_provided` is not needed, because we can check it with a single access using the `application.cover_letter_file`. No need to wrap this single line in a property.

`is_cover_letter_provided` is mainly used, because it is using an `or` conditional to check for cover letter file or cover letter text. If either of them is present, then "COVER LETTER" would be a part of the fieldset title.